vllm/core/kv_cache_manager.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict, List, Optional, Set, Tuple
+
+import numpy as np
+
+from vllm.request import Request
+from vllm.logger import init_logger
+from vllm.utils import cdiv
+
+logger = init_logger(__name__)
+
+
+class KVCacheManager:
```

Comment:
Is this the new interface for the block manager? Should we implement prefix caching/hierarchical cache in this class?